File: src/lessons/maze/randommouse/RandomMouseMazeEntity.py
def run():
    import random

    def random3():
        return random.randrange(0,3)

    def setX(i):
            errorMsg("Sorry Dave, I cannot let you use setX(x) in this exercise. Walk to your goal instead.")
    def setY(i):
            errorMsg("Sorry Dave, I cannot let you use setY(y) in this exercise. Walk to your goal instead.")
    def setPos(x,y):
            errorMsg("Sorry Dave, I cannot let you use setPos(x,y) in this exercise. Walk to your goal instead.")

    # BEGIN SOLUTION
    while not isOverBaggle():
        n = random3()
        if n == 0:
            if not isFacingWall():
                forward()
        elif n == 1:
            left()
        else:
            right()
    pickupBaggle()
    # END SOLUTION

File: src/lessons/maze/randommouse/RandomMouseMazeEntity.java
package lessons.maze.randommouse;

public class RandomMouseMazeEntity extends plm.universe.bugglequest.SimpleBuggle {
	@Override
	public void setX(int i)  {
		if (isInited())
			throw new RuntimeException(getGame().i18n.tr("Sorry Dave, I cannot let you use setX(x) in this exercise. Walk to your goal instead."));
	}
	@Override
	public void setY(int i)  { 
		if (isInited())
			throw new RuntimeException(getGame().i18n.tr("Sorry Dave, I cannot let you use setY(y) in this exercise. Walk to your goal instead."));
	}
	@Override
	public void setPos(int i,int j)  { 
		if (isInited())
			throw new RuntimeException(getGame().i18n.tr("Sorry Dave, I cannot let you use setPos(x,y) in this exercise. Walk to your goal instead."));
	}

	/* BEGIN TEMPLATE */ 
	public void run() {
		// Your code here =)
		/* BEGIN SOLUTION */ 
		while (!isOverBaggle()) 
		{
			switch(random3()) 
			{
			case 0:
				if (!isFacingWall())
				{
					forward();
				}
				break;
			case 1:
				left();
				break;
			case 2:
				right();
				break;
			}
		}
		pickupBaggle();
		/* END SOLUTION */
	}
	/* END TEMPLATE */

	public int random3() {
		double n = Math.random();
		if (n < 0.33) {
			return 0;
		} else if (n < 0.66) {
			return 1;
		} else {
			return 2;
		}
	}

}
